iroh-net-report/src/reportgen/probes.rs

```diff
@@ -298,9 +313,11 @@ impl ProbePlan {
                         .expect("adding IcmpIpv6 probe to and IcmpIpv6 probe set");
                 }
             }
-            plan.add(https_probes);
-            plan.add(icmp_probes_ipv4);
-            plan.add(icmp_probes_ipv6);
+
+            plan.add_probes(
+                &protocols,
+                vec![https_probes, icmp_probes_ipv4, icmp_probes_ipv6],
+            );
```

Comment:
I wonder if you can add the `protocols` to the plan itself and `plan.add()` would then be able to check if it actually needs adding.  That way this code doesn't have to change (I don't know why, but the old version with 3 different calls to plan.add() is more readable to me).  Maybe rename it to `plan.add_if_enabled(probes)`.

Reply:
agreed